Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -0,0 +1,766 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ * Copyright (c) 2021, Stephan Unverwerth <s.unverwerth@gmx.de>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "SoftwareGLContext.h"
+#include "GLStruct.h"
+#include "SoftwareRasterizer.h"
+#include <AK/Assertions.h>
+#include <AK/Debug.h>
+#include <AK/Format.h>
+#include <AK/QuickSort.h>
+#include <AK/Vector.h>
+#include <LibGfx/Bitmap.h>
+#include <LibGfx/Painter.h>
+#include <LibGfx/Vector4.h>
+#include <math.h>
+
+using AK::dbgln;
+
+namespace GL {
+
+static constexpr size_t NUM_CLIP_PLANES = 6;
+
+static FloatVector4 clip_planes[] = {
+    { -1, 0, 0, 1 }, // Left Plane
+    { 1, 0, 0, 1 },  // Right Plane
+    { 0, 1, 0, 1 },  // Top Plane
+    { 0, -1, 0, 1 }, // Bottom plane
+    { 0, 0, 1, 1 },  // Near Plane
+    { 0, 0, -1, 1 }  // Far Plane
+};
+
+static FloatVector4 clip_plane_normals[] = {
+    { 1, 0, 0, 1 },  // Left Plane
+    { -1, 0, 0, 1 }, // Right Plane
+    { 0, -1, 0, 1 }, // Top Plane
+    { 0, 1, 0, 1 },  // Bottom plane
+    { 0, 0, -1, 1 }, // Near Plane
+    { 0, 0, 1, 1 }   // Far Plane
+};
```

Comment:
Please mark these const.

Also, I'm not sure if this would be a micro-opt, but you could remove the clip_plane_normals array, since `clip_plane_normals[i] == clip_planes[i^1]`.